.github/workflows/nightly_dependency_tests.yaml

```diff
@@ -0,0 +1,76 @@
+name: Nightly dependencies@develop
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 23 * * *"
+
+concurrency:
+  # github.workflow: name of the workflow, so that we don't cancel other workflows
+  # github.event.pull_request.number || github.ref: pull request number or branch name if not a pull request
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  # Cancel in-progress runs when a new workflow with the same group name is triggered
+  # This avoids workflow runs on both pushes and PRs
+  cancel-in-progress: true
+
+jobs:
+  style:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.12
+
+      - name: Check formatting with pre-commit
+        run: |
+          python -m pip install pre-commit
+          pre-commit run ruff
+
+  Tests:
+    needs: style
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
+        # Include MacOS M-series Runners
+        include:
+          - os: macos-14
+            python-version: "3.10"
+          - os: macos-14
+            python-version: "3.11"
+          - os: macos-14
+            python-version: "3.12"
```

Comment:
Question: do we need to test against all Python versions and all operating systems? It could soon become redundant, i.e., whether to drop support and bump the lower bound for a PyBaMM version is already being tested with the dynamic GitHub Actions matrix we created earlier, so we can use those tests to decide (we will be dropping just released versions of PyBaMM anyway, and testing develop-PyBaMM with something like Windows and Python 3.8 might not make sense. If it doesn't to you, too – we can consider keeping just `ubuntu-latest` + Python 3.12 and removing the matrix (and Python 3.13/dev if needed in a 1x2 matrix).

Reply:
That's a good point, I'll update this to be `ubuntu-latest` + 3.12.